httpcore/_backends/sync.py

```diff
@@ -156,12 +156,16 @@ def open_uds_stream(
         with map_exceptions(exc_map):
             sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
             sock.settimeout(connect_timeout)
-            sock.connect(path)
```

Comment:
you can still get a ResourceWarning if this socket is GCd after a KeyboardInterrupt

```
            try:
                sock.connect(path)
            except BaseException:
                sock.close()
                raise
```